datastore/datastore.go

```diff
@@ -116,5 +141,35 @@ func (ds *datastore) putObjectWithKey(kvObject KV, key ...string) error {
 	if kvObjValue == nil {
 		return errors.New("Object must provide marshalled data for key : " + Key(kvObject.Key()...))
```

Comment:
Let's adhere to start the error messages with lower case. Also please add some context.